src/rdf4cpp/rdf/Literal.cpp

```diff
@@ -1570,11 +1600,12 @@ Literal Literal::substr(size_t start, size_t len, Node::NodeStorage &node_storag
 
     auto const s = this->lexical_form();
 
-    if (start > s.size()) {
+    if (start > this->strlen()) {
         return Literal::make_string_like_copy_lang_tag("", *this, node_storage);
     }
 
-    auto const substr = s.view().substr(start, len);
+    auto substr = s.view() | una::ranges::views::utf8 | una::views::drop(start) | una::views::take(len) | una::ranges::to_utf8<std::string>();
```

Comment:
Can this somehow be converted to a `string_view` without an allocation? In theory it should work, not sure if the interface of `una` allows that tho.

Reply:
no